### PR TITLE
Add .npmignore

### PR DESCRIPTION
**Pull request by smithsz**
## What

Add `.npmignore` file.

```
$ npm pack
cloudant-couchbackup-2.0.0-SNAPSHOT.tgz
$ tar -ztvf cloudant-couchbackup-2.0.0-SNAPSHOT.tgz 
-rw-r--r-- 501/20         1429 2017-06-13 16:58 package/package.json
-rw-r--r-- 501/20          256 2017-06-14 15:19 package/.npmignore
-rw-r--r-- 501/20        10681 2017-06-08 17:18 package/README.md
-rw-r--r-- 501/20        11324 2017-05-16 15:15 package/LICENSE
-rw-r--r-- 501/20         8875 2017-06-12 17:38 package/app.js
-rw-r--r-- 501/20         1851 2017-06-12 17:38 package/CHANGES.md
-rwxr-xr-x 501/20         1999 2017-06-12 21:36 package/bin/couchbackup.bin.js
-rwxr-xr-x 501/20         1657 2017-06-08 07:52 package/bin/couchrestore.bin.js
-rw-r--r-- 501/20         8116 2017-06-13 13:41 package/includes/backup.js
-rw-r--r-- 501/20         2587 2017-06-01 11:50 package/includes/config.js
-rw-r--r-- 501/20         1338 2017-06-13 13:41 package/includes/error.js
-rw-r--r-- 501/20         1356 2017-05-31 15:41 package/includes/liner.js
-rw-r--r-- 501/20         1980 2017-06-08 14:55 package/includes/logfilegetbatches.js
-rw-r--r-- 501/20         1654 2017-06-01 11:50 package/includes/cliutils.js
-rw-r--r-- 501/20         3894 2017-06-07 10:55 package/includes/parser.js
-rw-r--r-- 501/20         1262 2017-06-13 13:41 package/includes/request.js
-rw-r--r-- 501/20         2194 2017-06-13 13:41 package/includes/restore.js
-rw-r--r-- 501/20         2170 2017-06-13 13:41 package/includes/shallowbackup.js
-rw-r--r-- 501/20         3015 2017-06-14 14:51 package/includes/spoolchanges.js
-rw-r--r-- 501/20         1273 2017-06-11 09:42 package/includes/change.js
-rw-r--r-- 501/20         6111 2017-06-13 13:41 package/includes/writer.js
-rw-r--r-- 501/20         2610 2017-06-09 02:22 package/includes/logfilesummary.js
```

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+.eslintrc*
+
+# tests
+citest
+test
+
+# test logs
+backup.txt
+npm-debug.log
+out
```
Comment (ricellis) [inline .npmignore]: Shouldn't we also ignore the `.npmignore` itself?

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# build tool config
+Jenkinsfile
+.travis.yml
+
+# code examples
+examples
+
+# docs
+CONTRIBUTING.md
+
```
Comment (ricellis) [inline .npmignore]: IIRC the `README` is displayed on the npm site, but I don't think `CHANGES` is used there perhaps we need not include that in the package either?

File: .npmignore (hunk 1 above)
Comment (smithsz) [inline .npmignore]: ah, will do

File: .npmignore (hunk 2 above)
Comment (smithsz) [inline .npmignore]: The npm docs say CHANGELOG (and its variants) are always included. Not sure why.
Even if I add it to .npmignore it still appears in the package.

Comment (ricellis) [inline .npmignore]: Fair enough, we'll leave it in. Perhaps it was/will be included in the npm generated pages at somepoint.

File: .npmignore (hunk 1 above)
Comment (smithsz) [inline .npmignore]: added fc85568942738d68faf09208cc9562613e837d02

**Pull request merged**